tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/lifecycle/AsyncInvoker.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.bulkhead.lifecycle;
+
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.testng.Assert.fail;
+
+@ApplicationScoped
+public class AsyncInvoker {
```

Comment:
I didn't know about `Exceptions.expect()`, it is indeed exactly the same and I'll use that.

When it comes to `AsyncCaller`, its `run` method (unlike `submit`) doesn't catch exceptins, and I'm not sure if that's intentional or mistake. So I created my own.

Reply:
Removed the assert method, but left `AsyncInvoker` for now.

Reply:
I had to look at this for a while before working out what I think is going on here.

I think `AsyncCaller.run()` doesn't catch exceptions because `@Asynchronous` will catch all thrown exceptions anyway.

`AsyncCaller.submit()` does catch exceptions because `Callable` throws checked exceptions so they have to be either caught or declared as thrown, even though `@Asynchronous` will always catch them.

Reply:
Ah, yea! Got it -- sorry it takes so long :-)